### DISCUSSION TITLE
Question about AI deck setup

### DISCUSSION THREAD

**Discussion by wdliu356** [bitcraze/discussions#1624] [Q&A]
I'm following the ai deck tutorial but met problem with getting the figure shown. I got some questions and list them below.
1. In the _Update Crazyflie and AIdeck firmware_ step, it says I should see `ESP32: I (910) SYS: Initialized` but what I saw is 
![Screenshot from 2024-11-01 11-12-07](https://github.com/user-attachments/assets/64c8575a-40e3-46eb-8809-d144feead3b9). The value in the bracket is slightly different. Does this mean the flashing work is not done successfully?

2. When I'm running the "wifi steaming example", I can only get `Connecting to socket on 192.168.4.1:5000...
Socket connected
` with no image received. It seems to get stuck at `data.extend(client_socket.recv(size-len(data)))`. Do you have any idea why this is happening?

3. I tried customizing the ai deck firmware following your documentation and I found the code stuck at `open_pi_camera_himax(&camera)`. I think this might also be the reason for the second question. Do you have any idea why this is happening?

**Comment by wdliu356** [comment]
Some update about my tests
Everything worked fine when I attached the AI deck to a normal crazyflie 2.1. But when I attached it to a bolt, the camera cannot be turned on. Any idea for this problem?

**Reply by gemenerik** [reply]
It sounds like it might be related to GAP8 initialization problems. I recommend implementing the workaround with a resistor https://github.com/bitcraze/aideck-gap8-examples/issues/105

Whether you attach it to a Bolt or regular CF2.1 shouldn't matter, could it be a coincidence?

**Reply by wdliu356** [reply]
I really don't think that's a coincidence since I have two ai decks and both worked with CF2.1 but not worked with the bolt. Very confusing. 
But I'll check the discussion you shared first. Thanks

**Reply by wdliu356** [reply]
> It sounds like it might be related to GAP8 initialization problems. I recommend implementing the workaround with a resistor [bitcraze/aideck-gap8-examples#105](https://github.com/bitcraze/aideck-gap8-examples/issues/105)

Hi, I don't see the point why you mention this discussion. I can flash the firmware without any error and it worked fine with a CF2.1. Even when attaching it to a bolt, the connection can be set up fine though the camera cannot be opened.

**Reply by wdliu356** [reply]
I switched to my second bolt and it works fine now. Seems to be an hardware issue.....

**Reply by wdliu356** [reply]
I did some more tests and it seems that when I connect my esc's to the bolt, then the camera cannot be turned on. If I unplug all the esc's, then everything works good. Looks like a powering issue? I'm currently using a 3S battery. Do you think that is enough for a bolt to run with an AI deck?

**Reply by gemenerik** [reply]
I referenced the previous discussion as this issue does appear power-related, which can sometimes impact the GAP8's initialization. The workaround in https://github.com/bitcraze/aideck-gap8-examples/issues/105 suggests placing a 4.5 kOhm resistor between VCOM and ground to help the GAP8 reset more reliably by allowing the voltage to drop quickly. This could be a straightforward way to check if the GAP8 reset is the source of the issue. If you have access to a resistor of this value or similar, you might try temporarily attaching it to see if the AI-deck and camera stabilize.

I don’t have enough expertise with hardware to confirm whether the connection to ESCs would amplify this effect or suggest that it’s unrelated to the GAP8 initialization issue. Perhaps @tobbeanton can step in.

**Reply by tobbeanton** [reply]
This could very much be the ESCs that holds energy in its capacitors causing the VCOM voltage to drop slower. A bleeding resistor as described above could solve the problem. Maybe a 1-2 kOhm resistor is needed depending on how large the ESC capacitors are.

**Reply by wdliu356** [reply]
Thanks for the advice. I'll get back to you as soon as I finish some further experiment.

**Reply by wdliu356** [reply]
Just wanna confirm you are talking about VCOM and ground pin on the bolt, right?

**Reply by wdliu356** [reply]
Hi, some more updates here.
I followed that repo and tried resistor from 1k ohm to 5k ohm and it did no help.

Other than that, I retested it with no esc's or motors connected and I found it can start successfully but also got stuck after some time.
I checked other posts and found this post https://github.com/bitcraze/aideck-gap8-examples/issues/106. In my case, the code got stuck in the exactly same place `data.extend(client_socket.recv(size-len(data)))`. Such issue also happens when using a normal crazyflie 2.1 but the connection can "survive" longer than using a bolt. I tested the voltage between the Vcom pin and the GND pin. On normal crazyflie, the voltage difference is about 3.5v and for bolt it is 4.8v (This is tested when connecting 4 esc's and motors). I don't see the reason why a higher voltage can lead to a worse performance.


I did some other interesting experiment. If I only connect one motor and esc to the board. It sometimes can start successfully and get stuck in about 30s. But if two pairs are connected, then the success rate is close to 0.

**Reply by wdliu356** [reply]
I did another test and makes me even more confused. I tried powering it with a constant DC power supply and had it work alone. I tried both 3.5v and 5v and none of of the voltage could turn on the camera successfully, stuck at `data.extend(client_socket.recv(size-len(data))).` I could see a current reading of 0.112A and the wifi signal was all good.

**Reply by wdliu356** [reply]
Another interesting result is that the transfer speed of JPEG_ENCODING is way quicker than RAW_ENCODING.

**Reply by wdliu356** [reply]
Some more test result. I added some log in the `wifi-streamer.c` file and found that in most cases, the camera_task got stuck at `open_pi_camera_himax(&camera)`. Sometimes, if I was lucky enough to get through that line, the task would get stuck at image capture process. So probably it is still the issue of the camera initialization.

About the transfer issue I mentioned yesterday, I think it might be the format problem. It is way easier to transfer JPEG than raw data.

**Reply by gemenerik** [reply]
Regarding JPEG transfer speeds, it makes sense to me that it’s faster than RAW.

Yes, you should connect the bleeding resistor between VCOM and ground. I’m sorry to hear it didn’t help, though. It’s also interesting that the bolt seems to worsen the issue.

If you want to investigate further, consider checking the voltage on VCOM (e.g. with an oscilloscope). It should ideally drop to pretty much zero before the deck starts. This is the reason for using the bleeding resistor. It’s possible that the ESCs add so much capacitance that this discharge takes longer. Again, my hardware knowledge is limited, but I hope this helps.

Thank you for sharing all this information. If you come across anything new, please let us know. We’ll need some time to look into this thoroughly, so it may take a while.
